plasma_framework/contracts/src/exits/payment/PaymentInFlightExitable.sol

```diff
@@ -108,6 +127,70 @@ contract PaymentInFlightExitable is
         emit InFlightExitStarted(msg.sender, startExitData.inFlightTxHash);
     }
 
+    function challengeInFlightExitNotCanonical(ChallengeCanonicityArgs memory args) public payable {
+        // Check if there is an active in-flight exit from this transaction?
+        uint192 exitId = ExitId.getInFlightExitId(args.inFlightTx);
+        PaymentExitDataModel.InFlightExit storage ife = inFlightExits[exitId];
+        require(ife.exitStartTimestamp != 0, "In-fligh exit doesn't exists");
+
+        // Check that the exit is active and in period 1
+        verifyFirstPhaseNotOver(ife);
+
+        // Check if exit's input was spent via MVP exit
+        // If I'm correct we no longer can check any of inputs was previously exited by Std exit
+        // RootChain sets 254-bit in the timestamp to check this
+        /// verifyInputNotSpent(ife);
+
+        // Check that two transactions are not the same
+        require(
+            keccak256(args.inFlightTx) != keccak256(args.competingTx),
+            "The competitor transaction is the same as transaction in-flight"
+        );
+
+        PaymentTransactionModel.Transaction memory inFlightTx = PaymentTransactionModel.decode(args.inFlightTx);
+
+        // Check that shared input owner signes competing transaction
+        require(
+            isSpendingConditionMet(
+                ife.inputs[args.inFlightTxInputIndex].outputGuard,
+                uint256(0), // should not be used
+                inFlightTx.inputs[args.inFlightTxInputIndex],
+                // can we trust the caller or we should extend storage for input outputTypes?
+                args.competingTxInputOutputType,
+                args.competingTx,
+                // also, shouldn't txType be hardcoded into this contract, as it servers as exit game for this particular tx type?
+                inFlightTx.txType,
```

Comment:
@pdobacz @boolafish: also this exit game seems to be just for Payment Tx - should we hardcode `TxType` in it then?

Reply:
oh ahh....I don't think we want to "hardcode" because we'd like to reuse the exit game for PaymentTx v2, v3....
But I just noticed that if we're relying on the tx type of decoded tx, Payment v2 tx can run through our checks in a Payment v1 Exit Game. Probably we should inject the tx type from constructor then and use that?